src/TRestRawBaseLineCorrectionProcess.cxx

```diff
@@ -80,21 +84,41 @@ TRestRawBaseLineCorrectionProcess::~TRestRawBaseLineCorrectionProcess() {
 
 void TRestRawBaseLineCorrectionProcess::Initialize() {
     SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
     fInputEvent = NULL;
     fOutputEvent = new TRestRawSignalEvent();
 }
 
-void TRestRawBaseLineCorrectionProcess::InitProcess() {
-    if (fSignalsRange.X() != -1 && fSignalsRange.Y() != -1) fRangeEnabled = true;
-}
-
 TRestEvent* TRestRawBaseLineCorrectionProcess::ProcessEvent(TRestEvent* evInput) {
-    fInputEvent = (TRestRawSignalEvent*)evInput;
+    fInputEvent = dynamic_cast<TRestRawSignalEvent*>(evInput);
+    fInputEvent->InitializeReferences(GetRunInfo());
+
+    if (fReadoutMetadata == nullptr) {
+        fReadoutMetadata = fInputEvent->GetReadoutMetadata();
+    }
+
+    if (fReadoutMetadata == nullptr) {
```

Comment:
This is what's causing the pipeline to fail. This metadata member will not always be present (it needs to be added by a separate process). We don't want to throw an error whenever it's missing. Only throw an error when it's missing and the user elected to remove some signals by type.